dateutil/parser.py

```diff
@@ -673,21 +697,16 @@ def _parse(self, timestr, dayfirst=None, yearfirst=None, fuzzy=False,
 
         res = self._result()
         l = _timelex.split(timestr)         # Splits the timestr into tokens
+        tokens = l # alias that is easier to grep
 
-        # keep up with the last token skipped so we can recombine
-        # consecutively skipped tokens (-2 for when i begins at 0).
-        last_skipped_token_i = -2
-        skipped_tokens = []
+        skipped_idxs = set()
```

Comment:
IIRC the main reason I used a set here was because of the `if idx-1 in skipped_idxs` check.

The secondary concern was idempotency.  I think that eventually we're going to have to do multiple passes or look-backs or something.

The version you posted doesn't work for me.   I'm assuming the `for idx in skipped_tokens:` is supposed to be `for idx in skipped_idxs:`.  If I make that change, then in py3 I get `['foo bar']` and in py2 I get an `AttributeError` because there is no `list.clear`.